test/common/testlib.py

```diff
@@ -1271,6 +1271,17 @@ def restore_file(self, path, post_restore_action=None):
         else:
             self.addCleanup(self.machine.execute, "rm -rf %s" % path)
 
+    def write_file(self, path, content):
+        '''Write a new file on primary machine
+
+        This is safe for @nondestructive tests, the file will be removed during cleanup.
+        '''
+        m = self.machine
+        m.write(path, content)
+
+        if self.is_nondestructive():
+            self.addCleanup(m.execute, "rm {0}".format(path))
```

Comment:
We may need to add `-f` at some point, when we use it in tests that remove the file. But we can do that if/when needed.